crates/noirc_evaluator/src/ssa_refactor/ir/dfg.rs

```diff
@@ -132,13 +137,18 @@ impl DataFlowGraph {
     }
 
     /// Inserts a new instruction at the end of the given block and returns its results
+    ///
+    /// TODO: This is doing more than just inserting an instruction and its results
     pub(crate) fn insert_instruction_and_results(
         &mut self,
         instruction: Instruction,
         block: BasicBlockId,
         ctrl_typevars: Option<Vec<Type>>,
     ) -> InsertInstructionResult {
         use InsertInstructionResult::*;
+        // TODO: Is simplify still inserting the instruction?
+        // TODO: or are we assuming that simplify will always return a ValueId that
+        // TODO: has already been inserted?
```

Comment:
Simplify may potentially simplify an instruction to an existing, known value (not necessarily a constant). E.g. `add x, 0` is simplified to `x`. So it is expected `simplify` will always return a valid ValueId if it returns `SimplifyResult::SimplifiedTo`.